### PR TITLE
Autocomplete

### LINKED ISSUE #43: Address autofill/validation and conversion to lat/lon
Two things:
1. Change the "location" input box in our RegisterPage to autocomplete addresses. You can search for a React library to do this -- try to find one with a lot of downloads and good reviews.
2. Validate this address in the server (probably in the `/auth/register` route) and geocode it (convert it to latitude and longitude) to store in our database. Again, feel free to use a library for the geocoding.

### PR DESCRIPTION
**Pull request by zaidmamsa1**
## Motivation

Implement autocompleting location on RegisterPage and ProfilePage and geocoding latitude and longitude coordinates.

Fixes #43 

## Summary of Changes

Feel free to use sentences or a bulleted list.

- Loaded Google Maps Javascript API library
- Created new component to autocomplete location search
- In RegisterPage, we geocoded coordinates of the address passed in by the component (child)
- In ProfilePage's edit view, we did the same
- Made it so that the location input changes when the searched address is clicked

## Testing

Explain how you tested these changes (unit tests, manual UI testing, etc.) or N/A if testing is not necessary (and explain why).

N/A

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@ node_modules/
 
 # misc
 .DS_Store
+# .env
```
Comment (rrrliu) [inline .gitignore]: why is this commented out?

File: .env [hunk 2]
```diff
@@ -0,0 +1 @@
+SESSION_ID = keyboard cat
```
Comment (rrrliu) [inline .env]: never check in .env files, please `git rm --cached` this file

**Review by rrrliu — CHANGES_REQUESTED** [review]
awesome!! the autocomplete + map works v nicely :^). couple things:
1. merge with `main` to resolve merge conflicts and so that the diff will be up-to-date
2. don't expose environment variables on github
3. can we have the font and styling of the autocomplete list match that of our app? it should be more similar to how our dropdown looks in Filter
image
filter dropdown:
image

File: client/src/components/Autocomplete/LocationSearchInput.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                const className = suggestion.active
+                  ? 'suggestion-item--active'
+                  : 'suggestion-item';
+
+                const style = suggestion.active
+                  ? { backgroundColor: '#fafafa', cursor: 'pointer' }
+                  : { backgroundColor: '#ffffff', cursor: 'pointer' };
+                return (
+                  <div
+                    key="loveliness"
```
Comment (rrrliu) [inline client/src/components/Autocomplete/LocationSearchInput.js]: lovely, but key should be unique for each element in a map. use the index as the key

**Review by leebianca — DISMISSED** [review]

**Review by rrrliu — APPROVED** [review]

**Pull request merged**